doc/source/io.rst

```diff
@@ -440,6 +440,45 @@ individual columns:
     Specifying ``dtype`` with ``engine`` other than 'c' raises a
     ``ValueError``.
 
+.. note::
+
+   Reading in data with mixed dtypes and relying on ``pandas``
+   to infer them is not recommended. In doing so, the parsing engine will
+   loop over all the dtypes, trying to convert them to an actual
+   type; if something breaks during that process, the engine will go to the
+   next ``dtype`` and the data is left modified in place. For example,
+
+   .. ipython:: python
+
+       from collections import Counter
+       df = pd.DataFrame({'col_1':range(500000) + ['a', 'b'] + range(500000)})
+       df.to_csv('foo')
+       mixed_df = pd.read_csv('foo')
```

Comment:
I think it would be nice to show the output of `mixed_df['col_1'].dtype` as well, which shows that you have an object dtype (which is used for mixed dtypes)